src/restruct_poc/beams_calcs.hpp

```diff
@@ -0,0 +1,760 @@
+#pragma once
```

Comment:
This file really should be several files to better organize things.

Reply:
On the one hand, it would be easier to read if broken into smaller files; on the other, it's nice to have all the functors and required functions in one place, even if it's a little long. Then again, I'm used to 2,000-5,000 line source files :)

Reply:
I don't mind keeping all beam related calculations in the same file, despite the rather long length. In addition to the reasons above, it makes for nice and short includes in the calling code.

We could probably improve the file name - would something like `beam_fea` make more sense?

Reply:
How about `beams_functors.hpp` since that is what the users of the header would be wanting to use. Then we could have have `beams_struct.hpp` for defining the beams data structure and `beams_functions.hpp` for functions that operate on the `Beams` structure. We'll move `Beams` initialization, `UpdateStates`, etc. to `beams_functions.hpp`.